### PR TITLE
[MXNET-531] Add download util

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Create a Download Utils for all examples.
@nswamy @yzhliu @andrewfayres 
Retry scheme added: #11797

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: scala-package/examples/src/main/scala/org/apache/mxnetexamples/util/DownloadUtil.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package org.apache.mxnetexamples.util
+
+import java.io.File
+import java.net.URL
+
+import org.apache.commons.io.FileUtils
+
+object DownloadUtil {
```
Comment (nswamy) [inline scala-package/examples/src/main/scala/org/apache/mxnetexamples/util/DownloadUtil.scala]: I think we don't need to create Separate classes for Utilities, we can put it under one class `Util`.  WDYT?

Comment (lanking520) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/util/DownloadUtil.scala]: I only see it being used in the example sections. The `Util` includes tools more for data processing

Comment (nswamy) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/util/DownloadUtil.scala]: if follow this convention we'll have to create a class for each Utility method, for example read image from file, reshape image(for the purpose of illustration), that would just create a bunch of classes within the Util package. Can we move this when the Util class becomes unmanageable?

**Comment by Roshrini** [comment]
@yzhliu @nswamy Can you review this PR and approve it if ready? tracking this one for MXNet 1.3 release. Thanks!

**Pull request merged**